### PR TITLE
Contacts

### PR DESCRIPTION
**Pull request by josephbridgwaterrowe**
Allow a user to manage multiple contacts for an Entity.

### CODE REVIEW

File: spec/controllers/contacts_controller_spec.rb [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+        contact_entry = double(:contact_entry, :to_h => {}, :valid? => true)
+        allow(CreateContact)
+          .to receive(:call) { context }
+        allow(ContactEntry)
+          .to receive(:new) { contact_entry }
+
+        post :create, :entity_id => entity.id, :contact_entry => {}
+      end
+
+      it 'responds with status equal to 302' do
```
Comment (michaelachrisco) [inline spec/controllers/contacts_controller_spec.rb]: I was going to recommend using: 
http://betterspecs.org/#contexts
But the respond_with method seems to be missing when I ran my own controller test.

File: spec/form_objects/entity_entry_spec.rb [hunk 2]
```diff
@@ -1,11 +1,12 @@
 require 'rails_helper'
 
-RSpec.describe NewSalesperson, :type => :model do
+RSpec.describe EntityEntry, :type => :model do
   subject do
```
Comment (michaelachrisco) [inline spec/form_objects/entity_entry_spec.rb]: You can make this into a one liner: 
```Ruby
subject { EntityEntry.new }
```

File: spec/interactors/create_contact_spec.rb [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+
+      it 'has set the contact' do
+        expect(context.contact).to be_present
+      end
+    end
+
+    describe Contact do
+      subject { context.contact }
+
+      its(:present?) { is_expected.to be_truthy }
```
Comment (michaelachrisco) [inline spec/interactors/create_contact_spec.rb]: I can see why you like its now. That is clean :)

File: spec/controllers/contacts_controller_spec.rb (hunk 1 above)
Comment (josephbridgwaterrowe) [inline spec/controllers/contacts_controller_spec.rb]: I'll give it a go.

File: spec/interactors/create_contact_spec.rb [hunk 4]
```diff
@@ -0,0 +1,78 @@
...
+      its(:originator) { is_expected.to eq(user.id.to_s) }
+    end
+  end
+
+  context 'invalid parameters' do
+    let(:first_name) { nil }
+
+    describe 'context' do
+      it 'is failure' do
+        expect(context).to be_a_failure
```
Comment (michaelachrisco) [inline spec/interactors/create_contact_spec.rb]:
```Ruby
it 'is failure' do
  is_expected.to be_a_failure
end
```

File: spec/controllers/contacts_controller_spec.rb (hunk 1 above)
Comment (michaelachrisco) [inline spec/controllers/contacts_controller_spec.rb]: My test looks like this:
```Ruby
      subject { response.status }
      it 'responds with 200' do
        get :index, :format => :json
        is_expected.to respond_with 200
      end
```
Just to give a little reference.

Comment (josephbridgwaterrowe) [inline spec/controllers/contacts_controller_spec.rb]:
```ruby
    describe 'POST create' do
      before do
        # blah blah blah

        post :create, :entity_id => entity.id, :contact_entry => {}
      end

      it { is_expected.to respond_with 302 }
      it { is_expected.to redirect_to entity_path(entity) }
      it 'flashes a success message' do
        expect(subject.request.flash[:notice]).to_not be_nil
      end
    end
```

The trick seems to be not to use `subject`.

File: spec/form_objects/entity_entry_spec.rb (hunk 2 above)
Comment (josephbridgwaterrowe) [inline spec/form_objects/entity_entry_spec.rb]: Ta.

File: spec/interactors/create_contact_spec.rb (hunk 4 above)
Comment (josephbridgwaterrowe) [inline spec/interactors/create_contact_spec.rb]: Shit, I missed this one... I'll add that now... are we good after this?

Comment (michaelachrisco) [inline spec/interactors/create_contact_spec.rb]: Indeed. Will merge after this one :)

**Pull request merged**